R/dbi-result.R

```diff
@@ -126,17 +126,54 @@ setMethod("dbGetRowsAffected", "OdbcResult",
   }
 )
 
+#' @param param_description A data.frame with per-parameter attribute
+#' overrides.  Argument is optional; if used it must have columns:
+#' * param_index Index of parameter in query ( beginning with 1 ).
+#' * data_type Integer corresponding to the parameter SQL Data Type.
+#'   See \code{\link{ODBC_TYPE}}.
+#' * column_size Size of parameter.
+#' * decimal_digits Either precision or the scale of the parameter
+#'   depending on type.
```

Comment:
```suggestion
#' @param param_description A data frame containing per-parameter attribute overrides.
#'   Optional argument that, if provided, must contain the following columns:
#'   \describe{
#'     \item{param_index}{Index of parameter in query (beginning with 1).}
#'     \item{data_type}{Integer corresponding to the parameter SQL Data Type.
#'       See [ODBC_TYPE()].
#'     \item{column_size}{Size of parameter.}
#'     \item{decimal_digits}{Either precision or the scale of the parameter
#'       depending on type.}
#'   }
```

* Transitions to `\describe` for data frame columns
* Links to `ODBC_TYPE()` with roxygen shorthand
* Refers to "data frame" rather than `data.frame` since tibbles and other data.frame subclasses are fine